### PR TITLE
fix(form-settings): Ensure prefixes works as described #3488

### LINKED ISSUE #3488: fix(form-settings): Ensure prefixes works as described
## User Story

As a `Give Admin` I want to be able to use the new Give 2.2 prefixes feature and for it to behave as expected based on the settings description.

Currently the description suggests several prefixes that don't seem to be available and there doesn't seem to be a way to add them either. 

## Current Behavior

I currently see the following description in the form settings "Name Title Prefix" option:

> Do you want to add a name title prefix dropdown field before the donor's first name field? This will display a dropdown with options such as Mrs, Miss, Ms, Sir, and Dr for donor to choose from.

But the only options available are `Mr | Mrs | Ms`
image

## Expected Behavior

I expect that inline documentation matches with the actual options available.

## Possible Solution
There are two options to resolve this:

1. Add the ability to add custom prefixes
2. Update the form display settings description to be more accurate. 

If choosing option 2, here's my suggested wording:

`Enable a prefix to appear before the donor name. Options are "Mr., Mrs., and Ms."

Additionally, there should always me a period (.) after each of those abbreviations. 

## Tasks

- [ ] Either fix the form setting description; or add the ability to add custom prefixes
- [ ] Ensure a period is placed after each existing prefix option
- [ ] Ensure localization of this description will still be rendered correctly. 
- [ ] If adding custom prefixes, ensure they render in the form settings correctly, and are output on the form correctly, and are passed to the donation meta and donor emails correctly

## Environment
[Give branch 2.2](https://github.com/WordImpress/Give/tree/release/2.2)

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR resolves #3488 

## How Has This Been Tested?
I've tested this with all the possible scenarios as per the screenshots below.

## Screenshots (jpeg or gifs if applicable):
**Donation Receipt**
image

**Admin Settings & Metabox Settings**
image

**Donation Detail Page**
image

**Donor Detail Page**
image

**Donor Listing Page**
image

**Donations Listing Page**
image

**Emails**
image

## Types of changes
Bug fix (a non-breaking change which fixes an issue) 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [X] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/admin/forms/class-metabox-form-data.php [hunk 1]
```diff
@@ -301,9 +301,9 @@ function get_settings() {
 							'style'         => 'width: 100%',
 							'wrapper_class' => 'give-hidden give-title-prefixes-wrap',
 							'options'       => array(
-								'Mr'   => __( 'Mr', 'give' ),
-								'Mrs'  => __( 'Mrs', 'give' ),
-								'Ms'   => __( 'Ms', 'give' ),
+								'Mr.'   => __( 'Mr.', 'give' ),
```
Comment (ravinderk) [inline includes/admin/forms/class-metabox-form-data.php]: @mehul0810 create a helper function with a filter for default title fixes and use this function here.

File: includes/admin/settings/class-settings-display.php [hunk 2]
```diff
@@ -118,9 +118,9 @@ public function get_settings() {
 							'wrapper_class' => 'give-hidden give-title-prefixes-settings-wrap',
 							'style'         => 'width: 30%',
 							'options'       => array(
-								'Mr'  => __( 'Mr', 'give' ),
-								'Mrs' => __( 'Mrs', 'give' ),
-								'Ms'  => __( 'Ms', 'give' ),
+								'Mr.'  => __( 'Mr.', 'give' ),
```
Comment (ravinderk) [inline includes/admin/settings/class-settings-display.php]: @mehul0810 create a helper function with a filter for default title fixes and use this function here.

File: templates/shortcode-receipt.php [hunk 3]
```diff
@@ -30,7 +30,7 @@
 
 // Update donor name, if title prefix is set.
 if ( ! empty( $user['title'] ) ) {
-	$full_name    = "{$user['title']}. {$user['first_name']} {$user['last_name']}";
+	$full_name    = "{$user['title']} {$user['first_name']} {$user['last_name']}";
```
Comment (ravinderk) [inline templates/shortcode-receipt.php]: @mehul0810 last name and title prefix are not required, so they can be empty, use time when you create name string.

**Comment by raftaar1191** [comment]
## Finding 

Need to add `Title Prefixes` in the export CSV section as well.

1. When exporting Donor you will see a full name column that should contain the title prefixes if they are enabled 
image


2. At the time of **exporting donation history** it should also contain the coumns `Title Prefixes`

**Comment by mehul0810** [comment]
@DevinWalker I have a couple of question-related to the implementation of title prefix in export. Check @raftaar1191 [comment](https://github.com/WordImpress/Give/pull/3501#issuecomment-405831175) above
1. Do we need to have a separate column to manage title prefix in CSV file?
2. What should be the name of the column? `Donor Title` or `Title Prefix`?
3. If we don't need a separate column for title prefix then how we will manage it as there are separate columns for donor first name and last name?

**Comment by raftaar1191** [comment]
@mehul0810 Travis is getting failed can you please check it out

**Comment by mehul0810** [comment]
@raftaar1191 Failing Unit tests are fixed now. Also, implemented title prefix within Export Donation History and Logs. Please review the PR and let me know your feedback. Thanks!

**Pull request merged**